File: js/task2/S2_ARR_01.js
function parseNumbers(arr) {
  return arr
      .map(item => Number(item.trim()))
      .filter(num => !Number.isNaN(num));
}

console.log(parseNumbers([" 1 ", "x", "2"]));  
console.log(parseNumbers(["10", " 20 ", "abc", "30.5"])); 
console.log(parseNumbers([])); 
console.log(parseNumbers(["1", "2", "3"]));

File: python/task2/S2_ARR_01.py
def parseNumbers(arr):
  result = []
  for item in arr:
      try:
          num = float(item.strip())
          result.append(num)
      except:
          pass
  return result

print(parseNumbers([" 1 ", "x", "2"]))
print(parseNumbers(["10", " 20 ", "abc", "30.5"]))
print(parseNumbers([]))
print(parseNumbers(["1", "2", "3"]))
